lib/matplotlib/__init__.py

```diff
@@ -230,27 +230,43 @@ def _ensure_handler():
     return handler
 
 
-def set_loglevel(level):
+def set_loglevel(level, *, ensure_handler=True):
     """
-    Set Matplotlib's root logger and root logger handler level, creating
-    the handler if it does not exist yet.
+    Configure Matplotlib's logging levels.
+
+    Matplotlib uses the standard library `logging` framework under the root
+    logger 'matplotlib'.  This is a helper function to:
+
+    - set Matplotlib's root logger level
+    - optionally set the root logger handler's level, creating the handler
+      if it does not exist yet
 
     Typically, one should call ``set_loglevel("info")`` or
     ``set_loglevel("debug")`` to get additional debugging information.
 
+    Users or applications that are installing their own logging handlers
+    may want to directly manipulate ``logging.getLogger('matplotlib')`` rather
+    than use this function.
+
     Parameters
     ----------
     level : {"notset", "debug", "info", "warning", "error", "critical"}
         The log level of the handler.
 
+    ensure_handler : bool
```

Comment:
I guess I am not sure why we would add this.  A naive enduser unfamiliar with logging won't know what this means, and a library author familiar with logging should know not to call this, particularly given the change to the docstring above.  If we do add this, the docstring below probably needs to be clarified.

Reply:
I agree; a documentation update should be sufficient:

- Add a new section for discussion of logging within Matplotlib - the two use cases as a library and as an application
- End users can call set_level()
- For library developers, state that Matplotlib is using the standard library `logging` and its singleton logger is called 'matplotlib' and in library their library code get the 'matplotlib' logger and and the desired handler.